sklearn/utils/tests/test_validation.py

```diff
@@ -1213,3 +1213,85 @@ def test_check_sparse_pandas_sp_format(sp_format):
     assert sp.issparse(result)
     assert result.format == sp_format
     assert_allclose_dense_sparse(sp_mat, result)
+
+
+def make_types_table():
+    pd = pytest.importorskip("pandas")
+
+    ntp = np.core.numerictypes.allTypes
+    inpt = [{'np_name': it[0], 'np_type': it[1]} for it in ntp.items()]
+    dall = pd.DataFrame(inpt)
+
+    dall.index = dall['np_name']
+    dall.drop(['generic', 'void0', 'void', 'flexible', 'integer',
+               'signedinteger', 'unsignedinteger', 'character',
+               'inexact', 'floating', 'number'], inplace=True)
+
+    dall = dall[(dall['np_type'] != np.complexfloating) &
+                (dall['np_type'] != np.complex128) &
+                (dall['np_type'] != np.complex256) &
+                (dall['np_type'] != np.complex64)]
+
+    dall['dtype'] = dall['np_type'].map(np.dtype)
+    dall['dtype_name'] = dall['dtype'].map(lambda x: x.name)
+
+    dall = dall.reindex(columns=['dtype_name', 'dtype', 'np_name', 'np_type'])
+    num_types = (lambda x: x[(x.dtype_name.str.startswith('bool')) |
+                 (x.dtype_name.str.startswith('float')) |
+                 (x.dtype_name.str.startswith('int')) |
+                 (x.dtype_name.str.startswith('uint'))
+                 ])(dall).copy()
+    num_types.sort_values(by='dtype_name', inplace=True)
+    num_types.index = list(range(num_types.shape[0]))
+
+    return num_types
+
+
+@pytest.mark.parametrize('dt_name',
+                         ['bool', 'float', 'int', 'uint'])
```

Comment:
We can explicitly list out the combinations we are about there

```py
@pytest.mark.parametrize('ntype1, ntype2', [
	("int32", "long"),
...
])
def test_check_pandas_sparse_invalid(ntype1, ntype2):
	pd = pytest.importskip("pandas")
	df = pd.DataFrame(...)
	with pytest.raises(ValueError, ...):
		check_array(df, ...)

@pytest.mark.parametrize('ntype1, ntype2', [
	("int", "long"),
...
])
def test_check_pandas_sparse_valid(ntype1, ntype2):
	# make sure valid combinations result in the expected sparse matrix.

Reply:
I changed the tests accordingly